[HO QI YUAN JANALD] iP

janald99 opened the pull request:
### **DUKE**

> "Tasks are tasks. You gotta do what you gotta do. But *automating* does what you gotta do for you." - *Janald Ho*

Duke helps you keep track of your tasks at hand, so that you don't have to worry about forgetting what you need to do. It's,
- text-based
- *easy* to use ☺️ 
-  ~~FAST~~ **SUPER FAST** to use 😍 
- most importantly, **__FREE__**!!! 🤑 

All you need to do is,

1. download it from [here](https://github.com/janald99/ip).
2. double-click it. 
3. add your tasks.
4. let it manage your tasks for you 😉 

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the ```main```  method:
```ruby
public class Main {
    public static void Main (String[] args) {
        new Duke("data/duke.txt").run();
    }
}
```

Commit:
Add Gradle support

janald99 committed:
Level 1. Greet, Echo, Exit.

janald99 committed:
Introduced OOP principles, progress on lvl 2

janald99 committed:
Level 2. Add, List

janald99 committed:
Level 3. Mark as Done

janald99 committed:
Added task file

janald99 committed:
Level 3. Mark as Done

janald99 committed:
Revert "Level 3. Mark as Done"

This reverts commit 35a4f9d1976b9e30ef2d0354023d1f8435fd208d.

janald99 committed:
Added task children classes: Deadline, Todo, Event, added toString() to Task

janald99 committed:
Level 4. ToDos, Events, Deadlines

janald99 committed:
Abstract actions into one method and clean main

janald99 committed:
Automated Text UI Testing. Also fixed string typos in Duke

janald99 committed:
Progress on handling exceptions, yet to organize them.

janald99 committed:
Level 5. Handle Errors

janald99 committed:
Level 6. Delete

janald99 committed:
Change to static

janald99 committed:
Edit exceptions, improve getFirstWord method

janald99 committed:
Fix switch and if statement indentations

janald99 committed:
Clean up Duke class and Update Expected.txt

janald99 committed:
Edit String[] inputWords so that it does not only split into 2

janald99 committed:
Level-7. Add storage class and Tasklist class to improve OOP

janald99 committed:
Handle FileNotFoundException

janald99 committed:
Level-8. Utilize LocalDateTime class to recognize deadline in Deadline

janald99 committed:
Merge branch 'branch-Level-7'

janald99 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

janald99 committed:
Change filepath

janald99 committed:
Ignore files in data

janald99 committed:
Add Ui class and change Storage constructor parameter

janald99 committed:
Improve OOP, Fix bug at Storage filepath.

janald99 committed:
Encapsulate more messages into Messages

janald99 committed:
Improve Ui and Message implementation

janald99 committed:
Abstracted reading strings to Parser

janald99 committed:
Merge branch 'branch-A-moreOOP'

# Conflicts:
#	.gitignore

janald99 committed:
Organize classes into packages

janald99 committed:
Add JUnit tests to Parser, Task, Todo classes, and remove TaskList field from Ui class

janald99 committed:
Add JUnit tests to Parser, Task, Todo classes, and Remove TaskList field from Ui class

janald99 committed:
Package the App as Jar file

janald99 committed:
Add JavaDocs

janald99 committed:
Comply to Java Coding Standards

janald99 committed:
Comply to switch statement coding style

janald99 committed:
Add find command for users to find a task by searching for a keyword

janald99 committed:
Merge branch 'branch-A-JavaDoc'

janald99 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/parser/Parser.java

janald99 committed:
Remove divider when there is InvalidCommand

blaketingyu commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+        while(!isExit) {
+            try {
+                  String fullCommand = ui.readCommand();
+                  ui.divide();    // show the divider line ("__________")
```
I like the way you use comments to make your code more readable!

blaketingyu commented on src/main/java/duke/parser/Action.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * Actions according to the available commands in Duke.
+ */
+public enum Action {
```
Good use of enum here! :D

blaketingyu commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,134 @@
+package duke.parser;
+
+import duke.command.AddCommand;
```
I like how you listed the imports lexicographically! 👍

blaketingyu commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+ * A type of Task, which has a deadline the task is due by.
+ */
+public class Deadline extends Task {
+
```
Maybe you could give each of your class attributes a short description. 👍

janald99 replied:
Thank you for your feedback! Would definitely add more descriptions in the next commits. 👍

blaketingyu commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+
+    protected String by;
+    private LocalDateTime dateTime;
+
```
Maybe you could give your Deadline constructor the method header! 👍

blaketingyu commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,28 @@
+ */
+public class Event extends Task{
+    protected String at;
+
```
Same as the deadline class, I think you could give your constructor a method header hehe 👍

blaketingyu commented on src/main/java/duke/ui/Messages.java:
```diff
@@ -0,0 +1,70 @@
+ */
+public class Messages {
+
+    public static final String LINE_SEPARATOR = System.lineSeparator();
```
I like how you use CAPS for the final variables! 👍

blaketingyu commented on src/test/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,46 @@
+class ParserTest {
+
+    @Test
+    void testParse() {
```
LGTM, I believe you've covered a lot of testing to me!

blaketingyu reviewed (commented):
LGTM overall, maybe just a feedback would be to give your methods their respective method headers. but I believe you just forgotten to comment for some of them. Nevertheless, great work! 👍

MechFroG88 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+    private static Ui ui;
+
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
```
I think that `this.` can be omitted in this case since it is unnecessary.

MechFroG88 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,86 @@
+            switch (action) {
+            case TODO:
+                try {
+                    String description = Parser.parseDescription(this.input);
+                    tasks.add(new Todo(description));
+                    ui.printTaskAdded(tasks);
+                    storage.save(tasks);
+                } catch (InvalidArgumentException e) {
+                    ui.showError(e.getMessage());
+                } catch (IOException e) {
+                    ui.showError(Messages.SAVE_ERROR);
+                } finally {
+                    break;
+                }
```
I think it would be better if you put this whole chunk as another function, so you would not have a very long switch case. (Since there might be more cases in the future)

janald99 replied:
Great idea 👍  will do it when I have the time!

MechFroG88 commented on src/main/java/duke/ui/Messages.java:
```diff
@@ -0,0 +1,70 @@
+/**
+ * A class that consists of messages that the Ui will show to the user as needed.
+ */
+public class Messages {
```
I like your idea of putting all constant strings into a class!

MechFroG88 commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,50 @@
+     * @param tasks TaskList that command is executed on.
+     *
+     * @param ui User interface that interacts with the user.
+     *
```
Based on the javadocs guide in module website, I think there shouldn't be new line between @param.

janald99 replied:
i agree with you! Thank you, will fix that!

MechFroG88 reviewed (commented):
Overall I think your code is very nicely written and neat!

Just some suggestions that you could possibly look into it.

janald99 committed:
Merge remote-tracking branch 'myfork/add-gradle-support' into add-gradle-support

janald99 committed:
Rename test/duke to test/java to match convention

janald99 committed:
Edited Code Style to comply to Coding Standards using CheckStyle

janald99 committed:
Edited more files to comply to Coding Standards

janald99 committed:
Edit javaDocs in Ui.java for showToUser method, which uses Varargs

janald99 committed:
Add GUI to Duke

janald99 committed:
Add GUI to Duke

janald99 committed:
Bye command: make platform exit

bye command does not end the program.

Add Platform.exit in handleUserInput() when user inputs "bye".

Adding this exit is better because the user can still input commands
after "bye" previously. Adding this function would terminate the program
when "bye" is executed.

janald99 committed:
showCommands(): make output more readable

showCommands()'s displayed message is unsightly and not user-friendly.

Edit showCommands() into for loop and change COMMANDS from type String
to String[]

Extracting COMMANDS into an array helps the showCommands() function to
print the list of commands to be updated when a new command is added
into the array, by using a for-loop.

janald99 committed:
Add assertions to Storage.load() and AddCommand switch default cases

janald99 committed:
Add SLAP to Parser and AddCommand

janald99 committed:
Extract method to check if a task number isOutOfBounds, change
String output to local variable instead of global for code quality,
Add SLAP to Storage class

janald99 committed:
Add SLAP to Storage class

janald99 committed:
Merge pull request #2 from janald99/branch-A-CodeQuality

Branch a code quality

janald99 committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/storage/Storage.java

janald99 committed:
Add one more assert and add "/" to at and by in Parser

janald99 committed:
Remove image background and Change HBox minHeight to USE_PREF_SIZE

janald99 committed:
Fix bug at Storage by adding a whitespace to split("\\| "),
Improve GUI appearance

janald99 committed:
Change bot name and Images of user and duke

janald99 committed:
Fix bug at Parser, description can now handle more than one word

janald99 committed:
Add HelpCommand, edit invalidCommand to ask user to try 'help'

janald99 committed:
Improve CodeQuality, follow CheckStyle

janald99 committed:
Delete unused variables

janald99 committed:
Add Ui.png, remove Application in Duke, add Title to MainWindow

janald99 committed:
Edit javaDoc for isOutOfBounds() method to conform to coding standard

janald99 committed:
Fix background image zooming in by adding styles.css

janald99 committed:
Add background color to textField

janald99 committed:
Improve FindCommand to ignore case of keyword

janald99 committed:
Add a User Guide to README.md

janald99 committed:
Merge branch 'master' into branch-A-Assertions

janald99 committed:
Merge pull request #1 from janald99/branch-A-Assertions

Add assertions to Storage.load() and AddCommand switch default cases

janald99 committed:
Remove comment in Storage

janald99 committed:
Update README.md

janald99 committed:
Update README.md

janald99 committed:
Update README.md

janald99 committed:
Fix link in README.md

janald99 committed:
Add FAQ to README.md

janald99 committed:
Set theme jekyll-theme-cayman

janald99 committed:
Update README.md

janald99 committed:
Merge remote-tracking branch 'myfork/master'

janald99 committed:
Fix image in User Guide

janald99 committed:
Change image in User Guide

janald99 committed:
Fix find command bug where find 0 provides unmarked tasks

janald99 committed:
Add line separate for invalid command message

janald99 committed:
Add icon to duke

janald99 committed:
Fix JavaFX to be able to be used by Mac and Linux
